core/src/identity.rs

```diff
@@ -309,37 +316,39 @@ impl TryFrom<keys_proto::PublicKey> for PublicKey {
     type Error = DecodingError;
 
     fn try_from(pubkey: keys_proto::PublicKey) -> Result<Self, Self::Error> {
-        let key_type = keys_proto::KeyType::from_i32(pubkey.r#type)
-            .ok_or_else(|| DecodingError::new(format!("unknown key type: {}", pubkey.r#type)))?;
+        use protobuf::Enum;
+
+        let key_type = keys_proto::KeyType::from_i32(pubkey.Type().value())
+            .ok_or_else(|| DecodingError::new(format!("unknown key type: {}", pubkey.Type().value())))?;
```

Comment:
Are we roundtripping here? 🤔

Reply:
Now that I'm looking at it `keys_proto::KeyType::from_i32(pubkey.Type().value()` will not fail. I'll add an `expect`.